cpp/basix/math.cpp

```diff
@@ -86,3 +86,30 @@ basix::math::solve(const xt::xtensor<double, 2>& A, const xt::xarray<double>& B)
   return _B;
 }
 //------------------------------------------------------------------
+bool basix::math::is_singular(const xt::xtensor<double, 2>& A)
```

Comment:
@IgorBaratta Is this a sensible way to check if a matrix is singular or should we be doing something else?

Reply:
I think it's ok for now. 
Maybe we can reintroduce a function to check for singularity using `DGECON` (condition number) and `DGETRF` (LU factorization) later, since it's more precise but it can be more complicated.